### PR TITLE
feat/initial_implementation

### PR DESCRIPTION
**Pull request by Lorenzo45**
https://app.asana.com/0/1128417577549588/1139118931272205

MAX Demo App for Unity
iOS Screenshot: ................................ Android Screenshot:

### CODE REVIEW

File: DemoApp/Assets/Scenes/InterstitialScene.unity.meta [hunk 1]
```diff
@@ -0,0 +1,7 @@
+fileFormatVersion: 2
```
Comment (santoshbagadi) [inline DemoApp/Assets/Scenes/InterstitialScene.unity.meta]: I think you can remove this `.meta` file.

File: DemoApp/Assets/Scripts/HomeScreen.cs [hunk 2]
```diff
@@ -1,18 +1,226 @@
...
+        MaxSdkCallbacks.OnInterstitialHiddenEvent += OnInterstitialDismissedEvent;
+
+        // Load the first interstitial
+        LoadInterstitial();
+    }
+
+    void LoadInterstitial()
+    {
+        MaxSdk.LoadInterstitial(InterstitialAdUnitId);
+        interstitialStatusText.text = "Loading...";
```
Comment (santoshbagadi) [inline DemoApp/Assets/Scripts/HomeScreen.cs]: nit: Let's swap the calls. Should first set the status label before doing the action.

Same for other events.

File: DemoApp/Assets/Scripts/HomeScreen.cs [hunk 3]
```diff
@@ -1,18 +1,226 @@
...
+    }
+
+    void ShowInterstitial()
+    {
+        if (MaxSdk.IsInterstitialReady(InterstitialAdUnitId))
+        {
+            MaxSdk.ShowInterstitial(InterstitialAdUnitId);
+            interstitialStatusText.text = "Showing";
+        }
```
Comment (santoshbagadi) [inline DemoApp/Assets/Scripts/HomeScreen.cs]: Might be worth updating the status text in the else case right? Like setting it to `Not ready` or something.

File: DemoApp/Assets/Scripts/HomeScreen.cs [hunk 4]
```diff
@@ -1,18 +1,234 @@
...
+        showRewardedButton.onClick.AddListener(ShowRewardedAd);
+        showBannerButton.onClick.AddListener(ToggleBannerVisibility);
+        mediationDebuggerButton.onClick.AddListener(MaxSdk.ShowMediationDebugger);
+
+        MaxSdkCallbacks.OnSdkInitializedEvent += (MaxSdkBase.SdkConfiguration sdkConfiguration) =>
+        {
+            // AppLovin SDK is initialized, configure and start loading ads
+            Debug.Log("MAX SDK Initialized");
+
+            if (sdkConfiguration.ConsentDialogState == MaxSdkBase.ConsentDialogState.Applies)
```
Comment (thomasmso) [inline DemoApp/Assets/Scripts/HomeScreen.cs]: We shouldn't demo user privacy related stuff here, neither iOS nor Android does it.

File: DemoApp/Assets/Scripts/HomeScreen.cs [hunk 5]
```diff
@@ -1,18 +1,234 @@
...
+                // No need to show consent dialog, proceed with initialization
+            }
+            else
+            {
+                // Consent dialog state is unknown. Proceed with initialization, but check if the consent
+                // dialog should be shown on the next application initialization
+            }
+
+            MaxSdk.SetHasUserConsent(true);
+            MaxSdk.SetIsAgeRestrictedUser(false);
```
Comment (thomasmso) [inline DemoApp/Assets/Scripts/HomeScreen.cs]: No no no.

File: DemoApp/Assets/Scripts/HomeScreen.cs [hunk 6]
```diff
@@ -1,18 +1,234 @@
...
+        MaxSdk.SetBannerBackgroundColor(BannerAdUnitId, Color.black);
+    }
+
+    private void ToggleBannerVisibility()
+    {
+        if (!isBannerShowing)
+        {
+            MaxSdk.ShowBanner(BannerAdUnitId);
+            isBannerShowing = true;
+            showBannerButton.GetComponentInChildren<Text>().text = "Hide Banner";
```
Comment (thomasmso) [inline DemoApp/Assets/Scripts/HomeScreen.cs]: I feel like there was a cleaner way to set button text but can't recall off the top of my head

File: DemoApp/Assets/Scripts/HomeScreen.cs (hunk 4 above)
Comment (thomasmso) [reply DemoApp/Assets/Scripts/HomeScreen.cs]: Reasoning being - different pubs in different locations will have different implementations - there's no one-size-fits all and we don't want to be liable for anything.

File: DemoApp/Assets/Scripts/HomeScreen.cs (hunk 5 above)
Comment (thomasmso) [reply DemoApp/Assets/Scripts/HomeScreen.cs]: If pubs just set to have user consent for all their traffic, they can point to our example and we can be liable.

File: DemoApp/Assets/Scripts/HomeScreen.cs (hunk 6 above)
Comment (Lorenzo45) [reply DemoApp/Assets/Scripts/HomeScreen.cs]: Everything I saw from Google was using this method or a similar one. If you know of a cleaner way I'll be happy to use it!

File: DemoApp/Assets/MaxSdk.meta [hunk 7]
```diff
@@ -0,0 +1,8 @@
+fileFormatVersion: 2
```
Comment (thomasmso) [inline DemoApp/Assets/MaxSdk.meta]: Doesn't this file come with the `MaxSdk.cs` file?

File: DemoApp/Assets/Scripts/HomeScreen.cs (hunk 6 above)
Comment (thomasmso) [reply DemoApp/Assets/Scripts/HomeScreen.cs]: yup fair

File: DemoApp/Assets/MaxSdk.meta (hunk 7 above)
Comment (Lorenzo45) [reply DemoApp/Assets/MaxSdk.meta]: Looks like this is a meta file for the MaxSdk as a whole, there's a separate `MaxSdk.cs.meta` for the `MaxSdk.cs` file.

**Pull request merged**